xarray/core/duck_array_ops.py

```diff
@@ -184,6 +184,9 @@ def cumulative_trapezoid(y, x, axis):
 def astype(data, dtype, **kwargs):
     if hasattr(data, "__array_namespace__"):
         xp = get_array_namespace(data)
+        if xp == np:
```

Comment:
The namespace for numpy is `numpy.array_api` and `np.array_api.astype` exists. It seems like this should just work.

Reply:
This is how our get_array_namespace works:
https://github.com/pydata/xarray/blob/dafd726c36e24ac77427513a4a149a6933353b66/xarray/core/duck_array_ops.py#L44-L48

Which usually works. But not for astype.

Using np.array_api doesn't work because you have to use np.array_api.Array instead of np.ndarray:

```python
import numpy.array_api as nxp
nxp.astype(np.array([1, 2,]), np.dtype(float))

Traceback (most recent call last):

  File "C:\Users\J.W\AppData\Local\Temp\ipykernel_8616\23329947.py", line 1, in <cell line: 1>
    nxp.astype(np.array([1, 2,]), np.dtype(float))

  File "C:\Users\J.W\anaconda3\envs\xarray-tests\lib\site-packages\numpy\array_api\_data_type_functions.py", line 20, in astype
    return Array._new(x._array.astype(dtype=dtype, copy=copy))

AttributeError: 'numpy.ndarray' object has no attribute '_array'
```

I found it simpler to just change astype here.